sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -297,77 +260,46 @@ def batch_event_handler(event):
-                    .filter(models.StatusTracking.file_path_to_create == file_path)
-                    .first()
-                )
+    # Get filename from batch job command
+    file_path_to_create = command[5]
+    # Get job status
+    job_status = (
+        models.Status.SUCCEEDED
+        if event["detail"]["status"] == "SUCCEEDED"
+        else models.Status.FAILED
+    )
 
-                # update three fields with updated information
-                result.status = models.Status.SUCCEEDED
-                result.job_definition = event["detail"]["jobDefinition"]
-                result.ingestion_date = get_file_creation_date(file_path)
-                session.commit()
-
-                # Then write to file catalog table
-                sci_file = ScienceFilepathManager(filename)
-                metadata_params = sci_file.get_file_metadata_params()
-                metadata_params["file_path"] = file_path
-                metadata_params["status_tracking_id"] = result.id
-                update_file_catalog_table(metadata_params)
-                # Send event from this lambda for Batch starter
-                # lambda
-                response = send_event_from_indexer(filename)
-                logger.info(f"Sent event to EventBridge - {response}")
-
-        except Exception as e:
-            logger.error(str(e))
-            return http_response(status_code=400, body=str(e))
-
-    elif event["detail"]["status"] == "FAILED":
-        try:
-            # Update only status table with Failed status
-            with Session(db.get_engine()) as session:
-                # Had to query this way because the select statement
-                # returns a RowProxy object when it executes it,
-                # not the actual StatusTracking model instance,
-                # which is why it can't update table row directly.
-                result = (
-                    session.query(models.StatusTracking)
-                    .filter(models.StatusTracking.file_path_to_create == file_path)
-                    .first()
-                )
+    with Session(db.get_engine()) as session:
+        # Had to query this way because the select statement
+        # returns a RowProxy object when it executes it,
+        # not the actual StatusTracking model instance,
+        # which is why it can't update table row directly.
+        result = (
+            session.query(models.StatusTracking)
+            .filter(models.StatusTracking.file_path_to_create == file_path_to_create)
+            .first()
+        )
 
-                # update two fields with updated information
-                result.status = models.Status.FAILED
-                result.job_definition = event["detail"]["jobDefinition"]
-                session.commit()
-        except Exception as e:
-            logger.error(str(e))
-            return http_response(status_code=400, body=str(e))
+        if result is None:
```

Comment:
If you try to insert into the db, does it fail and return the object that is already there because it wasn't unique? I'm wondering if you could try and insert the object into the database right away and then just update whatever object comes out of it rather than having to query() to find the object?